src/fs/rock/RockSwapDir.cc

```diff
@@ -882,8 +882,12 @@ Rock::SwapDir::writeCompleted(int errflag, size_t, RefCount< ::WriteRequest> r)
 }
 
 void
-Rock::SwapDir::indexingCompleted()
+Rock::SwapDir::remoteIndexingCompleted()
 {
+    assert(IamWorkerProcess());
+    if(!theFile || theFile->error())
+        return;
```

Comment:
It is not clear what the check condition means, what this check accomplishes (or why it is needed). Perhaps the code is missing a C++ comment to explain why storeRebuildComplete() should be protected from the checked conditions?